armory/scenarios/poisoning_obj_det.py

```diff
@@ -0,0 +1,529 @@
+            for box_idx, box in enumerate(bounding_boxes):
+                labels[box_idx] = box.label
+                boxes[box_idx, 0] = box.x1
+                boxes[box_idx, 1] = box.y1
+                boxes[box_idx, 2] = box.x2
+                boxes[box_idx, 3] = box.y2
+
+            aug_images.append(img)
+            aug_ydicts.append(
+                {"boxes": boxes, "labels": labels, "scores": np.ones_like(labels)}
+            )
+
+        return np.array(aug_images, dtype=np.float32), aug_ydicts
+
+    def filter_label(self, y):
+        # Remove boxes/labels from y if the patch wouldn't fit.
+        # If no boxes/labels are left, return None as a signal to skip this image completely.
+
+        new_y = {"boxes": [], "labels": [], "scores": []}
+
+        if len(y) > 1:
+            raise ValueError(
+                f"filter_label accepts lists of length 1, got length {len(y)}"
+            )
+        y = y[0]
+        for box, label in zip(y["boxes"], y["labels"]):
+            if (
+                box[2] - box[0] > self.patch_x_dim + 1
+                and box[3] - box[1] > self.patch_y_dim + 1
+                # TODO: remove +1 after ART 1.15.0 fixes a bug
+            ):
+                new_y["boxes"].append(box)
+                new_y["labels"].append(label)
+                new_y["scores"].append(1)
+
+            else:
+                self.n_boxes_removed_by_class[label] += 1
+
+        if len(new_y["labels"]) != len(y["labels"]):
+            self.n_images_affected += 1
+
+        if len(new_y["labels"]) > 0:
+            new_y["boxes"] = np.array(new_y["boxes"])
+            new_y["labels"] = np.array(new_y["labels"])
+            new_y["scores"] = np.array(new_y["scores"])
+            return [new_y]
+        else:
+            self.n_images_removed += 1
+            return None
+
+    def load_train_dataset(self, train_split_default=None):
+        """
+        Load and create in memory dataset
+            detect_poison does not currently support data generators
+        """
+        # This is inherited and modified slightly to apply image resizing and augmentation
+        # which in the future will be done by a preprocessor in ART.
+        if train_split_default is not None:
+            raise ValueError(
+                "train_split_default not used in this loading method for poison"
+            )
+        adhoc_config = self.config.get("adhoc") or {}
+        self.attack_variant = self.config["attack"]["kwargs"]["attack_variant"]
+        self.train_epochs = adhoc_config["train_epochs"]
+        self.fit_batch_size = adhoc_config.get(
+            "fit_batch_size", self.config["dataset"]["batch_size"]
+        )
+        self.config["dataset"]["batch_size"] = 1
+        # Set batch size to 1 for loading data, this simplifies filtering small boxes.
+        # Note, training is unaffected since self.fit_batch_size is already set above.
+
+        self.label_function = lambda y: y
```

Comment:
What is the purpose of this? I don't see any way for this to be set elsewhere, can likely remove this line and replace line 296 (the only place using this)
```
                            self.label_function(
                                aug_y_train[i : i + self.fit_batch_size]
                            ),
```
with
```
                            aug_y_train[i : i + self.fit_batch_size],
```
I've made a separate comment with this change.


```suggestion
```

Reply:
Nvmd I see it is used in https://github.com/twosixlabs/armory/blob/1b88fd21e27ddac619955dae30a6062a53c1c379/armory/scenarios/poisoning_witches_brew.py#L305

Reply:
That's witches brew, but actually it is used in the inherited function [filter_dataset](https://github.com/twosixlabs/armory/blob/1b88fd21e27ddac619955dae30a6062a53c1c379/armory/scenarios/poison.py#L271)